crates/portalnet/src/overlay/service/utils.rs

```diff
@@ -0,0 +1,9 @@
+use ethportal_api::SszEnr;
+use ssz::Encode;
+
+/// Limits a to a maximum packet size, including the discv5 header overhead.
+pub fn pop_while_ssz_bytes_len_gt(enrs: &mut Vec<SszEnr>, max_size: usize) {
+    while enrs.ssz_bytes_len() > max_size {
+        enrs.pop();
```

Comment:
Typo in comment. Plus, it feels strange to name the variable `enrs`, though I understand the type constrains it for now. It would be possible to make it generic on a type that implements ssz encoding, I suppose.
```suggestion
/// Limits to a maximum packet size, including the discv5 header overhead.
pub fn pop_while_ssz_bytes_len_gt(elements: &mut Vec<SszEnr>, max_size: usize) {
    while elements.ssz_bytes_len() > max_size {
        elements.pop();
```

No strong feelings on any of this except the docstring. :)

Reply:
```rust
/// Limits the elements count to a maximum packet size, including the discv5 header overhead.
pub fn pop_while_ssz_bytes_len_gt<SSZObject: Encode>(
    elements: &mut Vec<SSZObject>,
    max_size: usize,
) {
    while elements.ssz_bytes_len() > max_size {
        elements.pop();
    }
}
```

I ended up just making the function generic, and expanded on your comment suggestion a bit, as I was still a little confused reading it even after removing the `a`